newrelic/terraform-provider-newrelic
# feat(channels): add notifications channels provider

# Description

- Add notifications channels to the provider under the notifications folder.

## Type of change

Please delete options that are not relevant.


## Checklist:

Please delete options that are not relevant.


## How to test this change?

- Add `newrelic.tf` file under testing folder:
```terraform
terraform {
  required_providers {
    newrelic = {
      version = "~> 99.0.0"
      source  = "newrelic/newrelic"
    }
  }
}

provider "newrelic" {
  api_key = "<API_KEY>"
  alias = "nerdgraph"
  region = "US"
  account_id = "<ACCOUNT_ID>"
}

resource "newrelic_notification_channel" "linoy_test" {
  name = "linoy-terraform-channel-test"
  type = "WEBHOOK"
  destination_id = "<Destination_ID>"
  product = "IINT"
  properties {
    key = "payload"
    value = "{\n\t\"id\": \"test\"\n}"
    label = "Payload Template"
  }
}

```
- Change the accountId according to the account you have permissions to and add your API_KEY and destination id.
- Run `terrraform apply` and than type yes.
- Check if a new channel was created via NerdGraph `aiNotifications`.
